gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +164,233 @@ def _get_prior_measurements_in_world_frame(
+                cam_idx = measurement.i
+
+                if intrinsics[cam_idx] is None or wRi_list[cam_idx] is None:
+                    raise ValueError("Camera intrinsics or rotation cannot be None for input track measurements")
+
+                measurement_xy = intrinsics[cam_idx].calibrate(measurement.uv)
+                measurement_homog = Point3(measurement_xy[0], measurement_xy[1], 1.0)
+                w_cam_U_track = Unit3(wRi_list[cam_idx].rotate(Unit3(measurement_homog).point3()))
+
+                # Direction starts at camera, but first index is track_id.
+                landmark_directions[(track_id, cam_idx)] = w_cam_U_track
+        return landmark_directions
+
+    def __run_averaging(
+        self,
+        num_images: int,
+        w_i2Ui1_dict: RelativeDirectionsDict,
+        w_i2Ui1_dict_tracks: RelativeDirectionsDict,
+        wRi_list: List[Optional[Rot3]],
+        i2Ti1_priors: Dict[Tuple[int, int], PosePrior],
+        absolute_pose_priors: List[Optional[PosePrior]],
+        scale_factor: float,
+    ):
+        """Runs the averaging optimization.
+
+        Args:
+            num_images: number of images.
+            w_i2Ui1_dict: Unit directions from i2 to i1 in world frame indexed by (i1, i2).
+            w_i2Ui1_dict_tracks: Directions from camera to track in world frame indexed by (track_id, camera_id).
+            wRi_list: camera rotations in world frame.
+            i2Ti1_priors: relative pose priors.
+            absolute_pose_priors: absolute pose priors.
+            scale_factor: scale factor for the esimated translations.
+
+        Returns:
+            List of camera translations in world frame, with as many entries as the number of images.
+        """
+        logger.info(
+            "Using {} track measurements and {} camera measurements".format(len(w_i2Ui1_dict_tracks), len(w_i2Ui1_dict))
+        )
+
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(NOISE_MODEL_DIMENSION, NOISE_MODEL_SIGMA)
+        if self._robust_measurement_noise:
+            huber_loss = gtsam.noiseModel.mEstimator.Huber.Create(HUBER_LOSS_K)
+            noise_model = gtsam.noiseModel.Robust.Create(huber_loss, noise_model)
+
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, noise_model)
+
+        # Run the optimizer.
+        try:
+            algorithm = TranslationRecovery()
+            if len(i2Ti1_priors) > 0:
+                # scale is ignored here.
+                w_i1ti2_priors = self._binary_measurements_from_priors(i2Ti1_priors, wRi_list)
+                wti_initial = self.__get_initial_values(absolute_pose_priors)
+                wti_values = algorithm.run(w_i1Ui2_measurements, 0.0, w_i1ti2_priors, wti_initial)
+            else:
+                wti_values = algorithm.run(w_i1Ui2_measurements, scale_factor)
+        except TypeError as e:
+            # TODO(akshay-krishnan): remove once latest gtsam pip wheels updated.
```

Comment:
can we modify this to explain which pip wheel does not support? which does 4.2a7 support? https://pypi.org/project/gtsam/#history

Reply:
Noting as resolved, thanks